### PR TITLE
Fixes #324 Use branch name from config when creating PR

### LINKED ISSUE #324: Publishing using editorial workflow does not merge with correct branch when using Github backend
**- Do you want to request a *feature* or report a *bug*?**

Bug

**- What is the current behavior?**

Publishing using editorial workflow using the Github backend creates a PR with 'master' as the base branch, even though a different branch is specified in config.yml

**- If the current behavior is a bug, please provide the steps to reproduce.**

I am using netlify-cms to edit static page content generated and served by Github Pages (jekyll), meaning that source must be committed to the gh-pages branch of the hosted repo. So my [config.yml](http://www.eamonnbell.com/netlify-cms-test/admin/config.yml) begins as follows

~~~
backend:
    name: github
    repo: eamonnbell/netlify-cms-test
    branch: gh-pages

publish_mode: editorial_workflow

# etc
~~~

1. Sign into admin interface
2. Create post
3. Move from 'Draft' to 'Waiting to go live'
4. Publish
5. Programmatically generated PR changes from temporary branch to effect publication of draft post is merged with master branch of eamonnbell/netlify-cms-test

**- What is the expected behavior?**

Programmatically generated PR should merge changes from draft branch with the branch specified in config.yml

**- Please mention your node.js, and operating system version.**
Node: v.7.7.3
Linux x86_64  4.7.4-1-ARCH

### PR DESCRIPTION
**Pull request by josephearl**
**- Summary**

Fixes #324 - Publishing using editorial workflow does not merge with correct branch when using Github backend

**- Test plan**

Tested correct branch name from config is passed to `createPR` as the base branch.

**- Description for the changelog**

Pass the branch name from the config to `createPR` as the base branch.

### CODE REVIEW

**Comment by josephearl** [comment]
Another solution would be to have the default value in `createPR` be `this.branch` instead of `"master"`.

**Comment by josephearl** [comment]
I tried to write a test for this, but just `import API from "../API"` gave me the following error:

```
TypeError: Super expression must either be null or a function, not undefined
      
      at _inherits (src/backends/netlify-auth/API.js:1:1704)
      at src/backends/netlify-auth/API.js:3:30
      at Object.<anonymous> (src/backends/netlify-auth/API.js:74:23)
      at Object.<anonymous> (src/backends/netlify-auth/implementation.js:5:38)
      at Object.<anonymous> (src/backends/backend.js:243:433)
      at Object.<anonymous> (src/valueObjects/AssetProxy.js:43:151)
      at Object.<anonymous> (src/backends/github/API.js:5:45)
      at Object.<anonymous> (src/backends/github/__tests__/API.spec.js:1:166)
      at process._tickCallback (internal/process/next_tick.js:109:7)
```

which seemed very weird

**Comment by josephearl** [comment]
Even weirder, 

```
import AssetProxy from "../../../valueObjects/AssetProxy";
import API from "../API";
```

does work!

**Comment by josephearl** [comment]
Added a test (which mocks the GitHub API - https://github.com/netlify/netlify-cms/issues/330)

**Comment by josephearl** [comment]
Updated snapshots

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "file-loader": "^0.8.5",
     "identity-obj-proxy": "^3.0.0",
     "imports-loader": "^0.6.5",
-    "jest-cli": "^16.0.1",
+    "jest-cli": "19.1.0-alpha.eed82034",
```
Comment (Benaiah) [inline package.json]: Why are we updating this and `jest` below to a specific alpha version? This looks like it was an unintentional change - if so, can you remove this from the commit?

File: yarn.lock [hunk 2]
```diff
@@ -59,11 +59,11 @@ JSONStream@^0.8.4:
     jsonparse "0.0.5"
     through ">=2.2.7 <3"
 
-abab@^1.0.0:
+abab@^1.0.3:
```
Comment (Benaiah) [inline yarn.lock]: Are these changes coming from the updated `jest` and `jest-cli` dependencies?

File: src/backends/github/__tests__/API.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import AssetProxy from "../../../valueObjects/AssetProxy";
```
Comment (Benaiah) [inline src/backends/github/__tests__/API.spec.js]: Fantastic - mocking the GH API is an existing issue (https://github.com/netlify/netlify-cms/issues/330). This is a great step towards that, which will let us add a test suite for the editorial workflow.

File: src/components/EntryEditor/__tests__/__snapshots__/EntryEditorToolbar.spec.js.snap [hunk 4]
```diff
@@ -1,3 +1,5 @@
-exports[`EntryEditorToolbar should disable and update label of Save button when persisting 1`] = `"<div><button disabled=\"\" class=\"\" type=\"button\" data-react-toolbox=\"button\">Saving...</button> <button class=\"\" type=\"button\" data-react-toolbox=\"button\">Cancel</button></div>"`;
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Benaiah) [inline src/components/EntryEditor/__tests__/__snapshots__/EntryEditorToolbar.spec.js.snap]: These snapshot changes shouldn't be a part of this PR - it looks like these are due to either the `jest` update above or a previous commit not updating the snapshots when they should have been. Either way, this is unrelated to the rest of the PR, and should probably be handled separately.

**Comment by josephearl** [comment]
@Benaiah the jest updates were because '.resolve' for testing promises was only introduced to jest recently (and will be released in 20.0.0). The snapshot updates were because jest 19+ has a new escaping syntax for snapshots.

**Comment by josephearl** [comment]
I'll see if there's an easy way of testing promises with an earlier version of jest

**Comment by Benaiah** [comment]
@josephearl I see. As far as I can tell, though, that version notation will keep us on 19.1.0-alpha even after 20 is released - is that the case?

File: package.json (hunk 1 above)
Comment (calavera) [inline package.json]: as long as the tests pass, it's fine. But we should update it as soon as there is a stable version.

**Comment by josephearl** [comment]
@Benaiah I see your point, yes I believe it will, let me fix that

**Comment by josephearl** [comment]
@Benaiah I've rebased and updated the `jest` and `jest-cli` version ranges in `package.json`.

File: yarn.lock (hunk 2 above)
Comment (josephearl) [inline yarn.lock]: Yes, the `abab` dependency comes from `jsdom` which is used by `jest`.

**Comment by josephearl** [comment]
@Benaiah let me if that's OK - or if we should look at keeping on a release version of jest and using another method to test the promise.

File: src/backends/github/API.js [hunk 5]
```diff
@@ -386,7 +386,7 @@ export default class API {
     return this.deleteRef("heads", branchName);
   }
 
-  createPR(title, head, base = "master") {
+  createPR(title, head, base = this.branch) {
```
Comment (erquhart) [inline src/backends/github/API.js]: Wondering if that `Super expression` error you saw was related to this use of `this`.

Comment (josephearl) [inline src/backends/github/API.js]: Just tested -- I still get the error if I change it back to `base = "master"` and use `this.createPR(options.commitMessage, branchName, this.branch)`, so it seems not

File: package.json (hunk 1 above)
Comment (erquhart) [inline package.json]: I'd second that it's fine, mostly because it's a dev dependency. Let's keep an eye out to update soon.

**Comment by Benaiah** [comment]
@josephearl the consensus is that it's fine - we'll just want to update to a stable version as soon as it's available.

**Comment by josephearl** [comment]
@Benaiah :+1: rebased

**Comment by Benaiah** [comment]
@josephearl [Jest has released v19.0.2](https://github.com/facebook/jest/releases) - could you update this PR to use the non-alpha version?

**Comment by josephearl** [comment]
@Benaiah this is using the alpha of v19.**2.0**, not v19.**0.2**

**Comment by Benaiah** [comment]
@josephearl oh, my bad. I'll give it one more look over and merge then. Thanks for the contribution!

**Comment by calavera** [comment]
conflict on yarn.lock 😭

**Comment by josephearl** [comment]
@calavera conflict resolved :smile:

**Comment by calavera** [comment]
Awesome! Thanks 🎉
